From 2a6fa90fe380ca85f237be681fa3dacdac4c821e Mon Sep 17 00:00:00 2001
From: Robby Cai <robby.cai@nxp.com>
Date: Tue, 28 Mar 2017 18:40:56 +0800
Subject: [PATCH 1701/5242] MLK-14518-2 pxp: set data path for pxp after reset

commit  859328d871fd711c1459e5e381ce0b3bdc7dad7b from
https://source.codeaurora.org/external/imx/linux-imx.git

it's needed for pxp legacy process, so set it just after
pxp does the reset.

Signed-off-by: Robby Cai <robby.cai@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/dma/pxp/pxp_dma_v3.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/dma/pxp/pxp_dma_v3.c b/drivers/dma/pxp/pxp_dma_v3.c
index 0854757..5acbbd3 100644
--- a/drivers/dma/pxp/pxp_dma_v3.c
+++ b/drivers/dma/pxp/pxp_dma_v3.c
@@ -5226,6 +5226,8 @@ static int pxp_probe(struct platform_device *pdev)
 	device_create_file(&pdev->dev, &dev_attr_block_size);
 	pxp_clk_enable(pxp);
 	pxp_soft_reset(pxp);
+	if (pxp->devdata && pxp->devdata->pxp_data_path_config)
+		pxp->devdata->pxp_data_path_config(pxp);
 	/* enable all the possible irq raised by PXP */
 	__raw_writel(0xffff, pxp->base + HW_PXP_IRQ_MASK);
 
@@ -5323,6 +5325,8 @@ static int pxp_resume(struct device *dev)
 	pxp_clk_enable(pxp);
 	/* Pull PxP out of reset */
 	pxp_soft_reset(pxp);
+	if (pxp->devdata && pxp->devdata->pxp_data_path_config)
+		pxp->devdata->pxp_data_path_config(pxp);
 	/* enable all the possible irq raised by PXP */
 	__raw_writel(0xffff, pxp->base + HW_PXP_IRQ_MASK);
 	pxp_clk_disable(pxp);
-- 
1.7.9.5

